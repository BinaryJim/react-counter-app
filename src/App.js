import React from "react";
import { CounterStateContext, CounterDispatchContext } from "./store";
import { useCounterStore } from "./store";
import { CounterContainer } from "./components/counter-container";
import { CounterOutput } from "./components/counter-output";
import { CounterActions } from "./components/counter-actions";
import "./scss/styles.scss";

export default function App() {
  const [count, actions] = useCounterStore();

  return (
    <CounterStateContext.Provider value={count} >
      <CounterDispatchContext.Provider value={actions} >
        <div className="app">
          <CounterContainer>
            <CounterOutput />
            <CounterActions />
          </CounterContainer>
        </div>
      </CounterDispatchContext.Provider>
    </CounterStateContext.Provider>
  );
}
