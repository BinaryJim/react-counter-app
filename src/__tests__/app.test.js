import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import App from "../App";

jest.mock("../store", () => ({
  ...jest.requireActual("../store"),
  useCounterStore: jest.fn().mockReturnValue({
    count: "test count",
    actions: "test actions",
  }),
}));

// Test render
describe("App - test render", () => {
  it("should render without crashing", () => {
    shallow(<App />);
  });
});

// Testing output of the render
describe("App - render", () => {
  let component;

  beforeEach(() => {
    component = shallow(<App />);
  });

  it("should match snapshot", () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it("should pass the correct count value via the 'CounterStateContext.Provider'", () => {
    expect(component.find("ContextProvider").at(0).prop("value")).toEqual(
      "test count"
    );
  });

  it("should pass the correct actions values via the 'CounterDispatchContext.Provider'", () => {
    expect(component.find("ContextProvider").at(1).prop("value")).toEqual(
      "test actions"
    );
  });
});
