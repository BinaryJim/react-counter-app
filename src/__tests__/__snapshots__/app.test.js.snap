// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App - render should match snapshot 1`] = `
<ContextProvider
  value="test count"
>
  <ContextProvider
    value="test actions"
  >
    <div
      className="app"
    >
      <CounterContainer>
        <CounterOutput />
        <CounterActions />
      </CounterContainer>
    </div>
  </ContextProvider>
</ContextProvider>
`;
